### PR TITLE
wavesurfer.js: Add backend setPeaks definition

### PR DESCRIPTION
**Pull request by rijk**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by caugner** [comment]
@rijk Thank you for your PR! 👏 

I have a couple of remarks:

- Please fill in the PR template (by checking the todo items that you have done).
- Please prefix the PR title with `wavesurfer.js: `
- Please provide a link to the WaveSurfer.js API docs or code referencing the `setPeaks` method.

File: types/wavesurfer.js/index.d.ts
Comment (caugner) [reply types/wavesurfer.js/index.d.ts:477]: The `setPeaks()` is actually already defined in `WebAudioBackend`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/813a8799e465a7d5f0d6776643f20f93681e85e4/types/wavesurfer.js/index.d.ts#L416

The `WaveSurfer.backend` property is actually of type `WebAudio  | MediaElement | MediaElementWebAudio`, which all extend `WebAudioBackend`. So we should deprecate/remove the `WaveSurferBackend` type and update the  `WaveSurfer.backend` type accordingly.

Would you like to do that?

**Review by caugner — CHANGES_REQUESTED** [review]
We should probably abandon the `WaveSurferBackend` type and use `WebAudio | MediaElement | MediaElementWebAudio` instead.

File: types/wavesurfer.js/index.d.ts
Comment (rijk) [reply types/wavesurfer.js/index.d.ts]: Hi @caugner , thanks and sorry I didn't put more effort in this PR, I was in the middle of some complex changes in my codebase. Removing `WaveSurferBackend` sounds good, would there still be a way to define the backend type so we get intellisense autocompletion on `wavesurfer.backend.`?

Comment (caugner) [reply types/wavesurfer.js/index.d.ts]: We could also just replace the `interface WaveSurferBackend { }` block by `type WaveSurferBackend = WebAudio | MediaElement | MediaElementWebAudio` and intellisense should work without problems.

Comment (caugner) [reply types/wavesurfer.js/index.d.ts]: That said, the `duration` parameter should be marked as optional in the `abstract class WebAudioBackend`.

**Review by caugner — APPROVED** [review]

File: types/wavesurfer.js/index.d.ts
Comment (caugner) [reply types/wavesurfer.js/index.d.ts]: FYI @rijk I have incorporated these changes in my PR which includes a big refactoring:

* https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52726/files#diff-8994b04bfcb8248c8c366e4119c6674b3fff11142c7318a46700b6e88acca98fR24
* https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52726/files#diff-3e1b7906c2401ddb2ac8e4c232e3db232d3923ec24ae0715507ca6852e2490fcR7
* https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52726/commits/60b2d0f66a790dd55aabf45e25c132475174d624

**Comment by elibarzilay** [comment]
(Standard rant: if you add tests, you wouldn't need for a maintainer approval.)

**Comment by caugner** [comment]
Ready to merge

**Comment by rijk** [comment]
Thanks for the help @caugner!

**Pull request merged**
